crates/contract/src/update.rs

```diff
@@ -223,6 +224,30 @@ impl ProposedUpdates {
         }
     }
 
+    /// Removes votes from the specified accounts.
+    pub fn remove_votes(&mut self, accounts_to_remove: &[AccountId]) {
+        accounts_to_remove
+            .iter()
+            .for_each(|account| self.remove_vote(account));
+    }
+
+    /// Removes votes from accounts that are not participants.
+    pub fn remove_non_participant_votes(&mut self, participants: &Participants) {
+        let non_participants: Vec<AccountId> = self
+            .vote_by_participant
+            .keys()
+            .filter(|voter| !participants.is_participant(voter))
+            .cloned()
+            .collect();
+
+        self.remove_votes(&non_participants);
+    }
+
+    /// Returns all account IDs that have voted.
+    pub fn voters(&self) -> Vec<AccountId> {
```

Comment:
this method is currently only used in tests. So we either remove it or
```suggestion
    /// Returns all account IDs that have voted.
    #[cfg(test)]
    pub fn voters(&self) -> Vec<AccountId> {
```